master/internal/api_user_intg_test.go

```diff
@@ -924,3 +926,207 @@ func getActivityEntry(ctx context.Context, userID model.UserID, entityID int32)
+		UserId: int32(userID),
+	})
+	require.NoError(t, err)
+	require.NotNil(t, token)
+
+	exists, err := getLongLivedTokensEntry(ctx, userID)
+	require.True(t, exists)
+	require.NoError(t, err)
+}
+
+func TestPostUserLongLivedTokenWithLifespan(t *testing.T) {
+	api, _, ctx := setupAPITest(t, nil)
+
+	userID, err := user.Add(
+		context.TODO(),
+		&model.User{
+			Username: uuid.New().String(),
+			Remote:   false,
+		},
+		nil,
+	)
+	require.NoError(t, err)
+
+	// With lifespan input
+	token, err := api.PostUserLongLivedToken(ctx, &apiv1.PostUserLongLivedTokenRequest{
+		UserId:   int32(userID),
+		Lifespan: lifespan,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, token)
+
+	err = testSetLifespan(ctx, t, userID, lifespan)
+	require.NoError(t, err)
+
+	exists, err := getLongLivedTokensEntry(ctx, userID)
+	require.True(t, exists)
+	require.NoError(t, err)
+}
+
+func TestGetLongLivedToken(t *testing.T) {
+	api, curUser, ctx := setupAPITest(t, nil)
+
+	// Create a test token without lifespan input
+	token, err := api.PostLongLivedToken(ctx, &apiv1.PostLongLivedTokenRequest{})
+	require.NoError(t, err)
+	require.NotNil(t, token)
+
+	tokenInfo, err := api.GetLongLivedToken(ctx, &apiv1.GetLongLivedTokenRequest{})
+	require.NoError(t, err)
+	require.NotNil(t, tokenInfo)
+	require.Equal(t, int32(curUser.ID), tokenInfo.LongLivedTokenInfo.UserId)
+}
+
+func TestGetUserLongLivedToken(t *testing.T) {
+	api, _, ctx := setupAPITest(t, nil)
+
+	userID, err := user.Add(
+		context.TODO(),
+		&model.User{
+			Username: uuid.New().String(),
+			Remote:   false,
+		},
+		nil,
+	)
+	require.NoError(t, err)
+
+	// Create a test token without lifespan input
+	token, err := api.PostUserLongLivedToken(ctx, &apiv1.PostUserLongLivedTokenRequest{
+		UserId: int32(userID),
+	})
+	require.NoError(t, err)
+	require.NotNil(t, token)
+
+	tokenInfo, err := api.GetUserLongLivedToken(ctx, &apiv1.GetUserLongLivedTokenRequest{
+		UserId: int32(userID),
+	})
+	require.NoError(t, err)
+	require.NotNil(t, tokenInfo)
+	require.Equal(t, int32(userID), tokenInfo.LongLivedTokenInfo.UserId)
+}
+
+func TestAuthzLongLivedToken(t *testing.T) {
+	api, authzUsers, curUser, ctx := setupUserAuthzTest(t, nil)
+
+	// POST API Auth check
+	expectedErr := status.Error(codes.PermissionDenied, "canCreateUsersOwnToken")
+	authzUsers.On("CanCreateUsersOwnToken", mock.Anything, curUser).
+		Return(fmt.Errorf("canCreateUsersOwnToken")).Once()
+
+	_, err := api.PostLongLivedToken(ctx, &apiv1.PostLongLivedTokenRequest{})
+	require.Equal(t, expectedErr.Error(), err.Error())
+
+	// GET API Auth check
+	expectedErr = status.Error(codes.PermissionDenied, "canCreateUsersOwnToken")
+	authzUsers.On("CanCreateUsersOwnToken", mock.Anything, curUser).
+		Return(fmt.Errorf("canCreateUsersOwnToken")).Once()
+
+	_, err = api.GetLongLivedToken(ctx, &apiv1.GetLongLivedTokenRequest{})
+	require.Equal(t, expectedErr.Error(), err.Error())
+}
+
+func TestAuthzUserLongLivedToken(t *testing.T) {
```

Comment:
It might be worth making this clear that this is more Administrative-oriented, it's not entirely clear what the difference between `TestAuthzLongLivedToken` and `TestAuthzUserLongLivedToken` is at a quick glance.